tests/FSharp.Compiler.ComponentTests/FSharpChecker/TransparentCompiler.fs

```diff
@@ -945,9 +877,9 @@ let ``TryGetRecentCheckResultsForFile returns result after edit of other file``
         tryGetRecentCheckResults "Second" expectSome // file didn't change so we still want to get the recent result
     } |> ignore
 
-[<Fact(Skip="Flaky. See https://github.com/dotnet/fsharp/issues/16766")>]
+[<Fact(Skip="TransparentCompiler assumeDotNetFramework differs from default checker")>]
```

Comment:
I was thinking to make this a Theory with both `assumeDotNetFramework` values and set them explicitly.

Reply:
TBH I have no recollection of why or when I wrote this skip msg 😅 It seems I had some understanding of it then by I have zero now 😄

Reply:
`assumeDotNetFramework` is an argument to `GetProjectSnapshotFromScript` which (for some, probably good, reason) has different default values in TC and BC. 

If we set it explicitly this test should pass. And we might as well check both cases. 

Anyway, it doesn't have to be solved in this PR.